### ISSUE TITLE
Officially support Typescript 3.5.x

### ISSUE THREAD

**Issue by danilofuchs** [typescript-eslint/typescript-eslint#577]
Microsoft recently deployed Typescript@3.5.0.
typescript-estree should be updated to handle new syntax changes.

Current error:
```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0

YOUR TYPESCRIPT VERSION: 3.5.1

Please only submit bug reports when using the officially supported version.

=============
```

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.6.0` |
| `TypeScript`                           | `3.5.1` |

**Comment by bradzacher** [comment]
We shall get around to it soon - it was released 24 hours ago.
We need to bump versions, and run test suites.

It's pretty straightforward to do, so anyone could do it and PR it in....

**Comment by amdrewnguyen** [comment]
@bradzacher I'm working on a PR for this, but there are some issues related to https://devblogs.microsoft.com/typescript/announcing-typescript-3-5/#breaking-changes that I'm trying to address. Is there a procedure for checking that a PR is alright outside of running the integration tests? I can add notes for contributing going forward as well.

**Comment by bradzacher** [comment]
Could you please clarify what you mean by "issues"?
What checks are failing?

Each project is an isolated unit, so you can run the tests for each individual package before moving on to the integration tests.

**Comment by amdrewnguyen** [comment]
`yarn run lint` output:

```bash
src/eslint-utils/deepMerge.ts:34:11 - error TS2536: Type 'string' cannot be used to index type 'T'.
34           acc[key] = deepMerge(first[key], second[key]);
             ~~~~~~~~
src/eslint-utils/deepMerge.ts:37:11 - error TS2536: Type 'string' cannot be used to index type 'T'.
37           acc[key] = second[key];
             ~~~~~~~~
src/eslint-utils/deepMerge.ts:40:9 - error TS2536: Type 'string' cannot be used to index type 'T'.
40         acc[key] = first[key];
           ~~~~~~~~
src/eslint-utils/deepMerge.ts:42:9 - error TS2536: Type 'string' cannot be used to index type 'T'.
42         acc[key] = second[key];
           ~~~~~~~~
Found 4 errors.
```

**Comment by bradzacher** [comment]
there are several commands that work within each of the repositories themselves.
`yarn typecheck`
`yarn build`
`yarn test`

Any of those will let you move package to package before going to the root and running them across the entire repo.

**Comment by gruckion** [comment]
What's the current state of this it's now July. Is there anything I can do to help?

**Comment by JamesHenry** [comment]
@gruckion The update was merged on 6 Jun as you can see above, and it was released to the latest tag on npm on the 9th of June: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v1.10.1

**Comment by gruckion** [comment]
> @gruckion The update was merged on 6 Jun as you can see above, and it was released to the latest tag on npm on the 9th of June: https://github.com/typescript-eslint/typescript-eslint/releases/tag/v1.10.1

Oh yea, sorry about that. Thank you for taking the time to respond to me. Thank you for all your hard work <3
